### ISSUE TITLE
[Windows simulator] frequently crashes when relaunching simulator

### ISSUE THREAD

**Issue by XeduR** [coronalabs/corona#186]
This has been going on for quite a while already, long before Corona transitioned to Solar2D.

When relaunching the simulator via `Ctrl+R`, the simulator frequently hangs for several seconds and then it crashes. The simulator console remains open and shows the following message: `The stdin connection has closed.`

I haven't been able to find any particular reason as to what triggers this. Sometimes I get a day or two by without a single incident, whereas other days the simulator seems to crash between ~5-10 relaunches.

There seem to be two different types of the error messages in Windows Event Viewer, which are:

```
Source: Application Error
Task Category: (100)
Event ID: 1000
Level: Error

Faulting application name: Corona Simulator.exe, version: 20.0.3619.0, time stamp: 0x5f6dd8b5
Faulting module name: ucrtbase.dll, version: 10.0.18362.1110, time stamp: 0x100b54ae
Exception code: 0xc0000409
Fault offset: 0x0009efbb
Faulting process ID: 0x3600
Faulting application start time: 0x01d6ac9bb559023c
Faulting application path: C:\Program Files (x86)\Solar2D\Corona Simulator.exe
Faulting module path: C:\Windows\System32\ucrtbase.dll
Report ID: f1a7cd7b-ec5a-492a-a666-0473356e0446
```

```
Source: Application Error
Task Category: (100)
Event ID: 1000
Level: Error

Faulting application name: Corona Simulator.exe, version: 20.0.3619.0, time stamp: 0x5f6dd8b5
Faulting module name: VCRUNTIME140.dll, version: 14.24.28127.4, time stamp: 0x5d8e68be
Exception code: 0xc000041d
Fault offset: 0x0000317e
Faulting process ID: 0x1a1c
Faulting application start time: 0x01d6a0c2c18bc38f
Faulting application path: C:\Program Files (x86)\Solar2D\Corona Simulator.exe
Faulting module path: C:\Windows\SYSTEM32\VCRUNTIME140.dll
Report ID: 75276772-80f8-44da-a270-d91e1b0ab2b3
```

```
Source: SideBySide
Task Category: None
Event ID: 78
Level: Error

Activation context generation failed for "C:\Program Files (x86)\Solar2D\Corona Simulator.exe".Error in manifest or policy file "" on line . A component version required by the application conflicts with another component version already active. Conflicting components are:. Component 1: C:\Windows\WinSxS\manifests\x86_microsoft.windows.common-controls_6595b64144ccf1df_6.0.18362.1082_none_e62ecab8f8d74ec7.manifest. Component 2: C:\Windows\WinSxS\manifests\amd64_microsoft.windows.common-controls_6595b64144ccf1df_6.0.18362.1082_none_9e8193e1e45b25c1.manifest.
```

Here's my development computer's information if that helps:
```
------------------
System Information
------------------
      Time of this report: 10/28/2020, 09:16:17
             Machine name: XEDHEAD-ORPHAN
               Machine Id: {FCD5D9DA-527D-49E9-9A9E-7789CAEC7611}
         Operating System: Windows 10 Home 64-bit (10.0, Build 18363) (18362.19h1_release.190318-1202)
                 Language: English (Regional Setting: English)
      System Manufacturer: MSI
             System Model: MS-7917
                     BIOS: V1.13 (type: UEFI)
                Processor: Intel(R) Core(TM) i5-4690K CPU @ 3.50GHz (4 CPUs), ~3.5GHz
                   Memory: 16384MB RAM
      Available OS Memory: 16332MB RAM
                Page File: 11219MB used, 37881MB available
              Windows Dir: C:\Windows
          DirectX Version: DirectX 12

---------------
Display Devices
---------------
           Card name: NVIDIA GeForce GTX 960
        Manufacturer: NVIDIA
           Chip type: GeForce GTX 960
            DAC type: Integrated RAMDAC
         Device Type: Full Device (POST)
          Device Key: Enum\PCI\VEN_10DE&DEV_1401&SUBSYS_85201043&REV_A1
       Device Status: 0180200A [DN_DRIVER_LOADED|DN_STARTED|DN_DISABLEABLE|DN_NT_ENUMERATOR|DN_NT_DRIVER] 
 Device Problem Code: No Problem
 Driver Problem Code: Unknown
      Display Memory: 10174 MB
    Dedicated Memory: 2007 MB
       Shared Memory: 8166 MB
```

**Comment by mark-sgs** [comment]
I think it is related to video.  Maybe related to [https://github.com/coronalabs/corona/issues/154](url)

**Comment by Shchvova** [comment]
A while ago I made a [build](https://github.com/Shchvova/corona/actions/runs/206742208) with new toolchain. Could you see if that would happen there too? It is a 3 month old test build, but should work out fine. Direct [download](https://github.com/Shchvova/corona/suites/1044833230/artifacts/13988947).

**Comment by XeduR** [comment]
I tried it out and after relaunching the simulator every few seconds, I got the familiar error report on 20th relaunch or so:

```
Faulting application name: Corona Simulator.exe, version: 20.0.9999.0, time stamp: 0x5f3505bb
Faulting module name: ucrtbase.dll, version: 10.0.18362.1110, time stamp: 0x100b54ae
Exception code: 0xc0000409
Fault offset: 0x0009efbb
Faulting process ID: 0x1660
Faulting application start time: 0x01d6aea67b6ed4dc
Faulting application path: C:\Program Files (x86)\Solar2D\Corona Simulator.exe
Faulting module path: C:\Windows\System32\ucrtbase.dll
Report ID: 5303705f-4a1e-46be-8c61-15ecae751370
Faulting package full name: 
Faulting package-relative application ID: 
```

```
Fault bucket 1613090007255619761, type 5
Event Name: BEX
Response: Not available
Cab Id: 0

Problem signature:
P1: Corona Simulator.exe
P2: 20.0.9999.0
P3: 5f3505bb
P4: ucrtbase.dll
P5: 10.0.18362.1110
P6: 100b54ae
P7: 0009efbb
P8: c0000409
P9: 00000007
P10: 

Attached files:
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER9D65.tmp.dmp
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER9F4A.tmp.WERInternalMetadata.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER9F5B.tmp.xml
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER9F59.tmp.csv
\\?\C:\ProgramData\Microsoft\Windows\WER\Temp\WER9F79.tmp.txt

These files may be available here:
\\?\C:\ProgramData\Microsoft\Windows\WER\ReportArchive\AppCrash_Corona Simulator_1b64e0909ddd6cdb78518de6891f2e81c822ce_22e9eab4_a65ee507-5122-403f-8cbf-d10b4df69b3f

Analysis symbol: 
Rechecking for solution: 0
Report Id: 5303705f-4a1e-46be-8c61-15ecae751370
Report Status: 268435456
Hashed bucket: 409aadc60b8d9782b662d8d0a93418b1
Cab Guid: 0
```

If it's of any use, here's a Report.wer file (zipped) of the event: [Report.zip](https://github.com/coronalabs/corona/files/5464521/Report.zip)

**Comment by sekodev** [comment]
Is it possible that those are related?

![crash](https://user-images.githubusercontent.com/5222368/99532024-b0ae5800-29b4-11eb-95dd-9d5a5c6c0cad.png)

**_How it happens:_**
- This message appears when the user tries to start another simulator instance while another one is already running.

**Comment by XeduR** [comment]
I've been working a little on a new project that is mostly just logic and very little graphics. I hadn't encountered any of these frequent crashes until I added the initial audio systems to the project, after which it began to occasionally crash again.

---

So, **my hunch is that this is somehow related to audio**.

I did some googling for "ucrtbase.dll audio" and I found plenty of promising forum threads, articles, etc. These crashes may be caused by audio drivers (like Realtek or Nahimic) on Windows with ASUS/MSI motherboards (I've got MSI).

**Comment by mark-sgs** [comment]
Since moving from MSI to LG Gram this problem has gone away.

Also regarding ucrtbase.dll crashes, I had a few yesterday for a coding error.  Try putting `print(os.date("%s"))` in main and you get a horrible crash.

**Comment by XeduR** [comment]
Yeah, funnily enough my desktop has MSI and laptop has ASUS, which explains why I've been experiencing these issues on both of my devices.

In order to reproduce these audio related crashes when Solar2D Simulator restarts, all you need to do is have audio files loaded via `audio.loadSound()`. There's no need to do anything else with the audio, just having them loaded will frequently crash the simulator when restarting the project.

---

As for your newfound crash, @mark-sgs, I'd recommend creating a new issue for it. That is weird indeed as it crashes immediately without raising any warnings or errors.

**Comment by XeduR** [comment]
@Shchvova 

I've managed to further isolate the problem and I've even prepared a sample project that demonstrates the crash (on affected devices). All that the sample project does is that it traverses a directory for audio files and then loads them using `audio.loadSound()`.

![image](https://user-images.githubusercontent.com/42606839/142414450-40242f34-aa63-4c83-b272-49d3c09aa581.png)

**The problem seems to lie entirely with mp3 files.** 

I tried the attached sample project with mp3, ogg and wav files. The files are all from the same source and I've just converted the original ogg files to wav and mp3. Seems like running the sample project with wav files takes about 20ms, ogg files takes about 60ms and mp3 takes around 120-150ms. 

There's no worry about the handling times as such, but when you run the sample project and assign the project to load audio files from any of the three folders, **the simulator seems to sporadically crash only when handling mp3 files**.

---

**How to replicate:**

As far as I know, this issue only affects Windows devices running MSI or ASUS motherboards. All that you need to do is open the attached sample project on such a device (other motherboards may be also affected) and ensure that on line 10, the folder is set to "mp3".

Then you just need to press Ctrl+R, wait a second or two and repeat until the simulator crashes. Sometimes it takes 3-5 tries and other times it takes 30-50, so it isn't crashing consistently.

**Comment by Shchvova** [comment]
This potentially could have been fixed by #414

**Comment by XeduR** [comment]
I ran my audioCrash project 300+ times now and then I left another project running with a macro spamming Ctrl+R for a good while. The simulator didn't crash after 1000+ restarts, so I gave up.

Seems like this issue has been fixed with [Solar2D 2022.3675](https://github.com/coronalabs/corona/releases/tag/3675).

Great work @ggcrunchy!
